addons/website/static/src/common/website_visitor_model.js

```diff
@@ -0,0 +1,35 @@
+import { fields, Record } from "@mail/core/common/record";
+
+export class WebsiteVisitor extends Record {
+    static _name = "website.visitor";
+    static id = "id";
+
+    country = fields.One("res.country", {
+        /** @this {import("models").WebsiteVisitor} */
+        compute() {
+            return this.partner_id?.country ?? this.country_id;
+        },
+    });
+    country_id = fields.One("res.country");
+    /** @type {string} */
+    display_name;
+    displayName = fields.Attr(undefined, {
+        compute() {
+            return (
+                this.partner_id?.name ||
+                this.partner_id?.display_name ||
+                this.display_name ||
+                `Visitor #${this.id}`
```

Comment:
Fallback value should be translatable. But can `display_name` from visitor actually be empty in practice?

Reply:
I didn't see any case in practice, but in one of the tests `Rendering of visitor banner`. Should we change the test or the calculation?

Reply:
Isn't the test just failing due to lack of data in test setup or missing return value in mock server then?

Reply:
Yes, I added the display name compute method to the mock model.